src/tests/ftest/util/dmg_utils.py

```diff
@@ -74,7 +75,61 @@ def storage_scan(self, verbose=False):
         self.set_sub_command("storage")
         self.sub_command_class.set_sub_command("scan")
         self.sub_command_class.sub_command_class.verbose.value = verbose
-        return self._get_result()
+        self.result = self._get_result()
+        self.log.info("--- self.result.stdout ---")
+        self.log.info(self.result.stdout)
+
+        if verbose:
+            vals = re.findall(
+                r"--------\n([a-z0-9-]+)\n--------|"
+                r"\n([a-z0-9_]+)[ ]+([\d]+)[ ]+([\d.]+) ([A-Z]+)|"
+                r"([a-f0-9]+:[a-f0-9]+:[a-f0-9]+.[a-f0-9]+)[ ]+"
+                r"(\S+)[ ]+(\S+)[ ]+(\S+)[ ]+(\d+)[ ]+([\d.]+)"
+                r"[ ]+([A-Z]+)[ ]*\n", self.result.stdout)
+            self.log.info("--- Verbose output parse result ---")
+            self.log.info(vals)
+
+            data = defaultdict(list)
+            data["host"] = vals[0][0]
+
+            i = 1
+            while i < len(vals[0]):
+                if vals[i][1] == "":
+                    break
+                data["scm_namespaces"].append(vals[i][1])
+                data["scm_socket_ids"].append(vals[i][2])
+                data["scm_capacity_vals"].append(vals[i][3])
+                data["scm_capacity_units"].append(vals[i][4])
+                i += 1
+
+            while i < len(vals):
+                data["pci_addrs"].append(vals[i][5])
+                data["models"].append("{} {}".format(vals[i][6], vals[i][7]))
+                data["fw_revisions"].append(vals[i][8])
+                data["nvme_socket_ids"].append(vals[i][9])
+                data["nvme_capacity_vals"].append(vals[i][10])
+                data["nvme_capacity_units"].append(vals[i][11])
+                i += 1
+
+            return data
```

Comment:
Ideally this method would have a single return statement.

Reply:
Done. Although not sure what's the difference.